src/s1reader/s1_orbit.py

```diff
@@ -56,16 +61,62 @@ def download_orbit(safe_file: str, orbit_dir: str):
-    orbit_file = os.path.join(orbit_dir, f"{orbit_dict['orbit_name']}.EOF")
-    if not os.path.exists(orbit_file):
-        download_orbit_file(orbit_dir, orbit_dict['orbit_url'])
+    if orbit_dict is not None:
+        orbit_file = os.path.join(orbit_dir, f"{orbit_dict['orbit_name']}.EOF")
+        if not os.path.exists(orbit_file):
+            download_orbit_file(orbit_dir, orbit_dict['orbit_url'])
 
-    return orbit_file
+        return orbit_file
+
+    # POEORB is not found, or there is no RESORB file that
+    # covers the sensing period + margin at the starting time.
+    # Try to find two subsequent RESORB files that covers the
+    # sensing period + margins at the startint time.
+    if orbit_dict is None:
+        pad_short = datetime.timedelta(seconds = PADDING_SHORT)
+        print('Attempting to download and concatenate RESORB files.')
+        orbit_dict_earlier = get_orbit_dict(mission_id,
+                                            start_time,
+                                            start_time + pad_short, 'AUX_RESORB')
+
+        orbit_dict_later = get_orbit_dict(mission_id,
+                                          start_time + margin_start_time - pad_short,
+                                          end_time + pad_short,
+                                          'AUX_RESORB')
+
+        orbit_dict_list = [orbit_dict_earlier, orbit_dict_later]
+
+        if orbit_dict_list:
+            orbit_file_indv_list = []
+            for orbit_dict in orbit_dict_list:
+                orbit_file = os.path.join(orbit_dir, f"{orbit_dict['orbit_name']}.EOF")
+                if not os.path.exists(orbit_file):
+                    download_orbit_file(orbit_dir, orbit_dict['orbit_url'])
+                orbit_file_indv_list.append(orbit_file)
+
+            # concatenate the RESORB xml file.
+            # NOTE Careful about the order how the RESORBs are concatenated
+            # to avoid the non-uniform spacing of OSVs during the sensing times
+            # 1111111111111111111111111                                    2222222222222222222222222
+            #                2222222222222222222222222      1111111111111111111111111
+            # 1111111111111112222222222222222222222222      1111111111111111111111111222222222222222
+            #                 |---sensing time---|                          |---sensing time---|
+            #    CASE 1: adding earlier RESORB to latter            CASE 2: Adding latter RESORB to earlier
+            #                                                       (non-uniform temporal spacing takes place
+            #                                                      between `1` and `2` during the sensing time)
+
+            # adding earlier RESORB to latter (i.e. CASE 1 above)
+            concat_resorb_file = combine_xml_orbit_elements(orbit_file_indv_list[1],
+                                                            orbit_file_indv_list[0])
+
+    return concat_resorb_file
```

Comment:
A couple questions
1. what does the `indv` mean in `orbit_file_indiv_list`? I can't immediately tell what that variable is going for
2. We have `if orbit_dict_list`, but that if statement is always `True`, whether or not `orbit_dict_earlier` or `orbit_dict_later` is None:

```python

In [1]: a = [None, None]

In [2]: if a:
   ...:     print("ok")
   ...:
ok
```
So maybe we should get rid of the `if`... but if we actually are testing something, we need an `else` because otherwise `concat_resorb_file` is not defined right now

Reply:
1. I meant 'individual` by `indv`, but I think we can just say like `orbit_file_indv_list` here.
2. Good point. Maybe I'll check `orbit_dict_list` with `all()`, and define the returning value with `else` statement.